force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -192,13 +210,22 @@ public with sharing class ClassesObjectsMethods {
     public static String getDayOfWeek(DateTime dt) {
         // Define a map of day of the week as a key and day number as the value
         // Monday = 1, Tuesday = 2, etc.
+        Map<String, Integer> daysOfTheWeek = new Map<String,Integer>();
 
-        // Get the day name using .format('EEEE');
+        daysOfTheWeek.put('Monday', 1);
+        daysOfTheWeek.put('Tuesday', 2);
+        daysOfTheWeek.put('Wednesday', 3);
+        daysOfTheWeek.put('Thursday', 4);
+        daysOfTheWeek.put('Friday', 5);
+        daysOfTheWeek.put('Saturday', 6);
+        daysOfTheWeek.put('Sunday', 7);
 
+        // Get the day name using .format('EEEE');
+        String dayOfWeek = dt.format('EEEE');
         // Get the day number from the map using the day name
-
+        Integer dayNumber = daysOfTheWeek.get(dayOfWeek);
         // Return the result as a string
-        return null; // Replace null with the variable you used to store the result
+        return String.valueOf(dayNumber); // Replace null with the variable you used to store the result
```

Comment:
This is a long winded question for practice, but could you do this in just 1 line by using the Datetime.format() method?